src/daft-core/src/array/ops/utf8.rs

```diff
@@ -479,6 +481,89 @@ impl Utf8Array {
         Ok(Utf8Array::from((self.name(), Box::new(arrow_result))))
     }
 
+    pub fn find(&self, substr: &Utf8Array) -> DaftResult<Int64Array> {
+        let self_arrow = self.as_arrow();
+        let substr_arrow = substr.as_arrow();
+        // Handle empty cases.
+        if self.is_empty() || substr_arrow.is_empty() {
+            return Ok(Int64Array::empty(self.name(), self.data_type()));
+        }
+        match (self.len(), substr.len()) {
+            // matching len case
+            (self_len, substr_len) if self_len == substr_len => {
+                let arrow_result = self_arrow
+                    .iter()
+                    .zip(substr_arrow.iter())
+                    .map(|(val, substr)| match (val, substr) {
+                        (Some(val), Some(substr)) => {
+                            Some(val.find(substr).map(|pos| pos as i64).unwrap_or(-1))
+                        }
+                        _ => None,
+                    })
+                    .collect::<arrow2::array::Int64Array>();
+
+                Ok(Int64Array::from((self.name(), Box::new(arrow_result))))
+            }
+            // broadcast pattern case
+            (self_len, 1) => {
+                let substr_scalar_value = substr.get(0);
+                match substr_scalar_value {
+                    None => Ok(Int64Array::full_null(
+                        self.name(),
+                        self.data_type(),
```

Comment:
Wait a minute.... Isn't `self.data_type()` a string in this case? Weird that this works at all, I think it's a bug?

Reply:
thanks for pointing out, think it's a bug.